include/votca/csg/beadstructure.h

```diff
@@ -279,12 +281,19 @@ std::vector<T *> BeadStructure<T>::getNeighBeads(Index index) {
 }
 
 template <class T>
-T *BeadStructure<T>::getBead(Index index) {
+T &BeadStructure<T>::getBead(const Index &index) {
+  assert(beads_.count(index));
+  return *beads_[index];
+}
+
+template <class T>
+const T &BeadStructure<T>::getBead(const Index &index) const {
   assert(beads_.count(index));
```

Comment:
maybe want to throw a message with the assert

Reply:
I'll just get rid of the method.